src/core/mac/mac_frame.cpp

```diff
@@ -703,7 +703,7 @@ uint8_t Frame::GetFooterLength(void)
 
 uint8_t Frame::GetMaxPayloadLength(void)
 {
-    return kMTU - (GetHeaderLength() + GetFooterLength());
+    return kMTU - (GetHeaderLength() + GetFooterLength() + 2);
```

Comment:
`GetFooterLength()` already accounts for the FCS, as can be seen on [L699](https://github.com/luwannest/openthread/blob/e8604318a4b6a796e30bfc993025b0a45acb91f6/src/core/mac/mac_frame.cpp#L699).